geo/src/algorithm/bool_ops/mod.rs

```diff
@@ -57,10 +57,20 @@ pub trait BooleanOps {
         &self,
         other: &impl BooleanOps<Scalar = Self::Scalar>,
         op: OpType,
+    ) -> MultiPolygon<Self::Scalar> {
+        self.boolean_op_with_fill_rule(other, op, FillRule::EvenOdd)
+    }
+
+    /// Performs a boolean operation with the specified fill rule.
+    fn boolean_op_with_fill_rule(
+        &self,
+        other: &impl BooleanOps<Scalar = Self::Scalar>,
+        op: OpType,
+        fill_rule: FillRule,
```

Comment:
Can you write a doc test verifying this works? I don't see where we're re-export the FillRule type. We don't want to require users to have to explicitly add i_overlay to their dependencies.